Release an arm64 / macOS apple silicon binary

The most recent release of zls (https://github.com/zigtools/zls/releases/tag/0.9.0) includes only x86 binaries. It'd be helpful for developers on the current generation of Apple computers to be able to use zls.

Could builds for arm64 macOS be included in the next release?

Sad, Zig(47c58cba5) cann't compile zls(latest ba68a4d)

```
 (master)$ ~/downloads/zig-macos-aarch64-0.10.0-dev.3024+47c58cba5/zig build -Drelease-safe
/Downloads/zig-macos-aarch64-0.10.0-dev.3024+47c58cba5/lib/zig/std/fs.zig:960:25: error: only 'IterableDir' can be iterated; 'IterableDir' can be obtained with 'openIterableDir'
    pub const iterate = @compileError("only 'IterableDir' can be iterated; 'IterableDir' can be obtained with 'openIterableDir'");
                        ^
error: zls...
error: The following command exited with error code 1:
/Downloads/zig-macos-aarch64-0.10.0-dev.3024+47c58cba5/zig build-exe /code/misc/zls/src/main.zig -OReleaseSafe --cache-dir /code/misc/zls/zig-cache --global-cache-dir /.cache/zig --name zls --pkg-begin build_options /code/misc/zls/zig-cache/options/ZN5T5qs__MtT8K02r_1S-BHCIQm36qquqqxMkxJvDpZqrwQ-wS_m2895sSwEs1B_ --pkg-end --pkg-begin known-folders /code/misc/zls/src/known-folders/known-folders.zig --pkg-end --enable-cache 
error: the following build command failed with exit code 1:
/code/misc/zls/zig-cache/o/f7ef6c24324123d6a9875be3d5afb20b/build /Downloads/zig-macos-aarch64-0.10.0-dev.3024+47c58cba5/zig /code/misc/zls /code/misc/zls/zig-cache /.cache/zig -Drelease-safe


```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -34,7 +34,7 @@ jobs:
       - name: Build artifacts
         if: ${{ matrix.os == 'ubuntu-latest' }}
         run: |
-          declare -a targets=("x86_64-windows" "x86_64-linux" "x86_64-macos" "i386-windows" "i386-linux")
+          declare -a targets=("x86_64-windows" "x86_64-linux" "x86_64-macos" "i386-windows" "i386-linux" "aarch64-macos")
           mkdir -p "artifacts/"
 
           for target in "${targets[@]}"; do

